internal/converter/converter.go

```diff
@@ -47,6 +47,10 @@ func GetOrCreateConverter(options *ast.Options) (message.Converter, error) {
 	if t == "" {
 		t = message.FormatJson
 	}
+	if t == message.FormatJson && len(options.Schema) > 0 {
```

Comment:
Maybe consider passing this schema to the converter instantiator to eliminate the if branch. So that future converter extensions can leverage the information as well. (Maybe a to-do item for v2?)

Reply:
we do conversion to convert number into string, thus we should allow the `typeNumber` for the string. That is to say, this is the part of the conversion job and worked in json converter layer here.